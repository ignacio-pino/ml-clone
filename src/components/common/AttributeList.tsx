import {
  Text,
  Table,
  TableContainer,
  Tbody,
  Tr,
  Td,
  Box,
} from "@chakra-ui/react";
import { IAttributeListData } from "../../Interfaces";

export default function AttributeList({
  attributeListData,
}: {
  attributeListData: IAttributeListData[];
}) {
  return (
    <Box>
      <Text as="b">Caracter√≠sticas</Text>
      <TableContainer mt={"3vh"}>
        <Table variant={"striped"} backgroundColor="gray.800" size="sm">
          <Tbody>
            {attributeListData.map(({ id, name, value_name }) => (
              <Tr key={id}>
                <Td fontWeight={"bold"}>{name}</Td>
                <Td>{value_name}</Td>
              </Tr>
            ))}
          </Tbody>
        </Table>
      </TableContainer>
    </Box>
  );
}
