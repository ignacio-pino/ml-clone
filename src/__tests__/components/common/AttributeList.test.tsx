import { render, screen } from "@testing-library/react";
import AttributeList from "../../../components/common/AttributeList";

describe("AttributeList Component", () => {
  const mockAttributeListData = [
    { id: "1", name: "Color", value_name: "Red" },
    { id: "2", name: "Size", value_name: "Medium" },
    { id: "3", name: "Material", value_name: "Cotton" },
  ];

  it("should match snapshot", () => {
    const container = render(
      <AttributeList attributeListData={mockAttributeListData} />
    ).container;
    expect(container).toMatchSnapshot();
  });

  it("renders attribute names and values correctly", () => {
    render(<AttributeList attributeListData={mockAttributeListData} />);

    const colorNameElement = screen.getByText("Color");
    const colorValueElement = screen.getByText("Red");
    const sizeNameElement = screen.getByText("Size");
    const sizeValueElement = screen.getByText("Medium");
    const materialNameElement = screen.getByText("Material");
    const materialValueElement = screen.getByText("Cotton");

    expect(colorNameElement).toBeInTheDocument();
    expect(colorValueElement).toBeInTheDocument();
    expect(sizeNameElement).toBeInTheDocument();
    expect(sizeValueElement).toBeInTheDocument();
    expect(materialNameElement).toBeInTheDocument();
    expect(materialValueElement).toBeInTheDocument();
  });

  it("renders the correct number of rows", () => {
    render(<AttributeList attributeListData={mockAttributeListData} />);
    const rows = screen.getAllByRole("row");
    expect(rows).toHaveLength(mockAttributeListData.length);
  });
});
