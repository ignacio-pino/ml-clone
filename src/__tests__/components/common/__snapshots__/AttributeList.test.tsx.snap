// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AttributeList Component > should match snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <b
      class="chakra-text css-0"
    >
      Caracter√≠sticas
    </b>
    <div
      class="chakra-table__container css-10e72oj"
    >
      <table
        class="chakra-table css-1i60z40"
      >
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <td
              class="css-in3yi3"
            >
              Color
            </td>
            <td
              class="css-0"
            >
              Red
            </td>
          </tr>
          <tr
            class="css-0"
          >
            <td
              class="css-in3yi3"
            >
              Size
            </td>
            <td
              class="css-0"
            >
              Medium
            </td>
          </tr>
          <tr
            class="css-0"
          >
            <td
              class="css-in3yi3"
            >
              Material
            </td>
            <td
              class="css-0"
            >
              Cotton
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
